File: algo-method.com/tasks/1103/main.py
N, K = map(int, input().split())
X = list(map(int, input().split()))


def func(min_dist: int) -> bool:
    k = 1
    now = X[0]
    for x in X[1:]:
        if min_dist <= x - now:
            k += 1
            if k == K:
                return True
            now = x
    return False


lo = 1
hi = 10**10
while hi - lo > 1:
    mid = (hi + lo) // 2
    if func(mid):
        lo = mid
    else:
        hi = mid

ans = lo
print(ans)

File: algo-method.com/tasks/1103/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const [N, K] = input().split(" ").map(Number);
const X = input().split(" ").map(Number);

const func = (minDist: number): boolean => {
  let k = 1;
  let now = X[0];
  for (const x of X.slice(1)) {
    if (minDist <= x - now) {
      if (++k === K) return true;
      now = x;
    }
  }
  return false;
};

let lo = 1;
let hi = 10 ** 10;
while (hi - lo > 1) {
  const mid = Math.floor((hi + lo) / 2);
  if (func(mid)) lo = mid;
  else hi = mid;
}

const ans = lo;
print(ans);
